### PR TITLE
Add a lit code improvement

### PR DESCRIPTION
**Pull request by devincd**
Signed-off-by: devincd <505259926@qq.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,3 +1,4 @@
+bazil.org/fuse v0.0.0-20160811212531-371fbbdaa898 h1:SC+c6A1qTFstO9qmB86mPV2IpYme/2ZoEQ0hrP+wo+Q=
```
Comment (howardjohn) [inline go.sum:1]: these changes seem unexpected?

Comment (devincd) [reply go.sum:1]: yes,

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 2]
```diff
@@ -685,7 +684,7 @@ func (cb *ClusterBuilder) applyConnectionPool(mesh *meshconfig.MeshConfig, mc *M
 		}
 	}
 
-	if settings != nil && settings.Http != nil && settings.Http.UseClientProtocol {
+	if settings.Http != nil && settings.Http.UseClientProtocol {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:687]: This is unrelated too, but is right.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**